code/modules/mob/living/silicon/ai/ai_say.dm

```diff
@@ -49,108 +49,137 @@
+		return
+	var/regex/check_for_bad_chars = regex("\[^a-zA-Z!?.,' :\]+")
+	if(check_for_bad_chars.Find(message))
+		to_chat(src, "<span class='notice'>These characters are not available on the 15.ai system: [english_list(check_for_bad_chars.group)].</span>")
 		return
+	last_announcement = message
 
 	announcing_vox = world.time + VOX_DELAY
 
-	log_game("[key_name(src)] made a vocal announcement with the following message: [message].")
-	log_talk(message, LOG_SAY, tag="VOX Announcement")
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
-
-	if(GLOB.vox_sounds[word])
-
-		var/sound_file = GLOB.vox_sounds[word]
-		var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
-		voice.status = SOUND_STREAM
-
-	// If there is no single listener, broadcast to everyone in the same z level
+	log_game("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	message_admins("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	play_vox_word(message, character_to_use, src, src.z, null)
+
+
+/proc/play_vox_word(message, character, mob/living/silicon/ai/speaker, z_level, mob/only_listener)
+	var/api_url = "https://api.15.ai/app/getAudioFile"
+	var/static/vox_voice_number = 0
+	var/datum/http_request/req = new()
+	vox_voice_number++
+	req.prepare(RUSTG_HTTP_METHOD_POST, api_url, "{\"character\":\"[character]\",\"text\":\"[message]\",\"emotion\":\"Contextual\"}", list("Content-Type" = "application/json", "User-Agent" = "/tg/station 13 server"), json_encode(list("output_filename" = "data/vox_[vox_voice_number].wav")))
+	req.begin_async()
+	UNTIL(req.is_complete())
+	var/datum/http_response/res = req.into_response()
+	if(res.status_code == 200)
+		var/full_name_file = "data/vox_[vox_voice_number].wav"
+		// Process the audio to match SS13 AI effects.
+		shell("./data/ffmpeg.exe \
+			    -nostats -loglevel 0 \
+			    -i ./[full_name_file] \
+			    -i ./sound/effects/SynthImpulse.wav \
+			    -i ./sound/effects/RoomImpulse.wav \
+			    -filter_complex \" \
+			        \[0\] apad=pad_dur=2 \[in_1\]; \
+			        \[in_1\] asplit=2 \[in_1_1\] \[in_1_2\]; \
+			        \[in_1_1\] \[1\] afir=dry=10:wet=10 \[reverb_1\]; \
+			        \[in_1_2\] \[reverb_1\] amix=inputs=2:weights=8 1 \[mix_1\]; \
+			        \[mix_1\] asplit=2 \[mix_1_1\] \[mix_1_2\]; \
+			        \[mix_1_1\] \[2\] afir=dry=10:wet=10 \[reverb_2\]; \
+			        \[mix_1_2\] \[reverb_2\] amix=inputs=2:weights=10 1 \[mix_2\]; \
+			        \[mix_2\] equalizer=f=7710:t=q:w=0.6:g=-6,equalizer=f=33:t=q:w=0.44:g=-10 \[out\]; \
+        			\[out\] alimiter=level_in=1:level_out=1:limit=0.5:attack=5:release=20:level=disabled\" \
+    			-vn -y ./data/vox_[vox_voice_number].mp3")
+
+		if (!istype(SSassets.transport, /datum/asset_transport/webroot))
+			log_game("CDN not set up, VOX aborted.")
```

Comment:
I feel like this should be higher up why give the prompt to users and download and remix the wav just to fail over a config issue. (but the check would also need to be here too since shell() sleeps and admins can disable the cdn.)